### ISSUE TITLE
filtering on sim.people.age + another field using '&' fails

### ISSUE THREAD

**Issue by RHull-IDM** [starsimhub/starsim#559]
This is a curious one. When the population changes due to births or deaths, filtering by sim.people.uid first in a sequence of conditions causes a shape mismatch ValueError. If the uid condition occurs after the other condition, it seems to work correctly.

```
import starsim as ss
import pandas as pd

ppl = ss.People(10000)
disease = ss.SIR()
pregnancy = ss.Pregnancy(pars={'fertility_rate': pd.read_csv(ss.root / 'tests/test_data/nigeria_asfr.csv')})
n_years = 3
sim = ss.Sim(dt=1 / 52, n_years=n_years, people=ppl, diseases=disease, demographics=pregnancy)
sim.run()

newborns = ((sim.people.age > 0) & (sim.people.uid > 10000)).uids # Works
newborns = ((sim.people.uid > 10000) & (sim.people.age > 0)).uids # Errors
```

**Comment by devclinton** [comment]
@RHull-IDM - I think this is related to what Robyn mentioned here - https://starsim-idm.slack.com/archives/C059G2HE57X/p1717649148257889? It seems like almost the same thing. I think Cliff and/or Romesh are looking at this behavior and https://github.com/starsimhub/starsim/issues/508?

**Comment by cliffckerr** [comment]
Closed by #561
